pymodbus/server/async_io.py

```diff
@@ -196,8 +209,9 @@ def execute(self, request, *addr):
                 traceback.format_exc(),
             )
             response = request.doException(merror.SlaveFailure)
+
         # no response when broadcasting
-        if not broadcast:
+        if not broadcast and response is not None:
```

Comment:
Why can response be None now, when it wasn't before ?

Reply:
I set it to None at the beginning, I'll clear it out, so it is closer to original code.